src/plone/restapi/serializer/dxcontent.py

```diff
@@ -84,6 +86,24 @@ def __call__(self, version=None, include_items=True):
                 value = serializer()
                 result[json_compatible(name)] = value
 
+        # Fetch discussion registry
+        registry = queryUtility(IRegistry)
+        settings = registry.forInterface(IDiscussionSettings, check=False)
+
+        # Check if discussion is allowed globally, overwritting whatever that
+        # is set in the object
+        if not settings.globally_enabled:
+            result['allow_discussion'] = False
+        else:
+            if 'allow_discussion' not in result or \
+               result.get('allow_discussion', False) is None:
+                # The object allow_discussion attribute is not set, then
+                # lookup for the default value on the fti
+                portal_types = getToolByName(self.context, 'portal_types')
+                document_fti = getattr(portal_types, self.context.portal_type)
+                result['allow_discussion'] = document_fti.getProperty(
+                    'allow_discussion')
+
```

Comment:
```
if not settings.globally_enabled:
    result['allow_discussion'] = False
elif result.get('allow_discussion', None) is None:
    # The object allow_discussion attribute is not set, then
    # lookup for the default value on the fti
    fti = portal_type[self.context.portal_type]
    result['allow_discussion'] = fti.getProperty('allow_discussion')
```

Reply:
@jaroel Finally changed the condition with your suggestion.